### PR TITLE
WT-13155 Permit WT_RESTART and WT_NOTFOUND return values in prefetch

### PR DESCRIPTION
**Pull request by mm-ng**

### CODE REVIEW

**Review by etienneptl — APPROVED** [review]
LGTM!

File: src/conn/conn_prefetch.c [hunk 1]
```diff
@@ -98,9 +98,17 @@ __prefetch_thread_run(WT_SESSION_IMPL *session, WT_THREAD *thread)
          */
         F_CLR_ATOMIC_8(pe->ref, WT_REF_FLAG_PREFETCH);
         (void)__wt_atomic_subv32(&((WT_BTREE *)pe->dhandle->handle)->prefetch_busy, 1);
-        WT_ERR(ret);
 
         __wt_free(session, pe);
+
+        /*
+         * Fix specific errors that prevented prefetch from making progress, they are harmless.
+         */
```
Comment (kommiharibabu) [inline src/conn/conn_prefetch.c:106]: Make it a single line comment.

Comment (kommiharibabu) [reply src/conn/conn_prefetch.c:106]: I feel `Fix` may not be the correct word here.  how about the `ignore`?

Comment (mm-ng) [reply src/conn/conn_prefetch.c:106]: @kommiharibabu s-all won't allow me to keep it as a single line comment, but I can change the wording of the comment.

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Pull request merged**